### ISSUE TITLE
Pre Release

### ISSUE THREAD

**Issue by metalvoidzz** [metalvoidzz/Sourcehold#9]
# Stuff to do:
- [ ] Working main menu
- [ ] Loading / Saving
- [ ] Placing buildings
- [ ] Simple AI / path finding
- [ ] Correct cutscene rendering
- [ ] Working user interface
- [ ] Better texture atlas algorithm

**Comment by FilipJanitor** [comment]
I would like to help. I could play around with the menus for example to get some feel for the codebase. Also, I hope the task is independent enough to not block others who would depend on its results while I am working on it. What do you think?
The AI looks interesting too, but it seems to depend on a lot of yet unimplemented parts - like tiles and units interfaces etc.

**Comment by metalvoidzz** [comment]
Yeah, there's a lot of unimplemented things ;-) Anyway, I would really appreciate any work on the UI code, since it's still pretty tedious to use (some helper classes would be nice!). I think it would help to create fairly small commits to make merging easier at this point.

**Comment by FilipJanitor** [comment]
Ok, I am sorry for polluting this thread but since I don't have any better idea on where to ask I will ask here - I started playing with the menus and got confused a bit.
The question I want to ask is about the texture atlas. I think it would be faster if you could just shortly sketch the basic idea behind it instead of me trying to guess it. What I dont understand is the mapping from files to entries - how do I know which file from the `data/` directory maps to some index? Currently I know which file does contain which picture. The next step for me is to map it to the entries - that means getting an entry index for some file.
Thanks!

**Comment by metalvoidzz** [comment]
No worries asking here ;-). The texture atlas is just a way to merge multiple textures from a GM1 file into a single larger one, since there are some files containing as much as 1000 entries. The way the atlas works is that you give it a number of images and their dimensions, and it will give you the position of the image on the larger atlas texture (the first image will stay at index 0 for example).

**Comment by metalvoidzz** [comment]
So, I actually thought of this a while back. When I started the project, I had dynamic resolution changes in mind, to support a better windowed mode or (maybe) even an Android version in the future. Now, I don't know if this would actually work when it comes to the UI, but maybe we could decide at runtime which of the 3 modes to use as a "base" and scale the UI accordingly. Let me know if this sounds dumb...

I guess adding the settings menu shouldn't be that hard as soon as we have a working menu system

**Comment by gloeyisk** [comment]
> # Stuff to do:
> * [x]  Windows compatibility
> * [x]  Working main menu
> * [x]  Working user interface
> * [ ]  Correct cutscene rendering
> * [ ]  Sound effects (ADPCM)
> * [ ]  Loading / Saving
> * [ ]  Placing buildings
> * [ ]  Simple AI / path finding

Will you add support for Stronghold Crusader?

**Comment by metalvoidzz** [comment]
Probably not. Maybe in a different repo when this one is playable.

**Comment by gloeyisk** [comment]
> Probably not. Maybe in a different repo when this one is playable.

Great, will be waiting!
